### PR TITLE
Training test

### PR DESCRIPTION
**Pull request by sjkoelle**
This adds a unit test for training a single small batch one gradient step, and also makes the landing places of some hyperparameters more explicit.

### CODE REVIEW

File: uberduck_ml_dev/trainer/tacotron2.py [hunk 1]
```diff
@@ -412,6 +408,8 @@ def initialize_loader(self, include_f0: bool = False, n_frames_per_step: int = 1
         return train_set, val_set, train_loader, sampler, collate_fn
 
     def train(self):
+        from datetime import datetime
```
Comment (zachwe) [inline uberduck_ml_dev/trainer/tacotron2.py:411]: Top of file

File: tests/tests/trainer/test_trainer.py [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
+        # NOTE (Sam): this number was taken from master on 8/24/22.
+        # Since folks are training on master right now, it is a reasonable benchmark
+        train_loss_start = 0.320
+        train_loss_4_datapoints_1_iteration = 0.319
+
+        assert math.isclose(lj_trainer.loss[0], train_loss_start, abs_tol=5e-4)
+
+        assert math.isclose(
+            lj_trainer.loss[1], train_loss_4_datapoints_1_iteration, abs_tol=5e-4
+        )
```
Comment (zachwe) [inline tests/tests/trainer/test_trainer.py:48]: Nice!

**Review by zachwe — APPROVED** [review]
